modules/ROOT/pages/streaming.adoc

```diff
@@ -12,6 +12,35 @@ Streaming supports efficient processing of large data objects such as files, doc
 * Simple routing rules based on message metadata are still possible
 * You can combine streaming and non-streaming endpoints
 
+== Data Streams Consumption
+To understand how data streams are consumed, review the following points:
+
+* Data streams cannot be consumed more than once
++
+In the following example, the flow shows the HTTP Listener source that receives a POST method with a body payload to write to the files. The flow writes the first file correctly, while the second file is created with empty content because each component that consumes a stream expects to receive a new stream. After the first File Write operation consumes the stream, the second File Write operation receives an empty stream. Thereby, the second operation has no content to write to a file.
++
+.Mule 3 Streaming: Writing a File
+image::mruntime-streaming-about-1.png[]
++
+In the following example, something similar happens when you try to log the payload before and
+after a DataWeave transformation. The flow logs the payload before the Transform Message component, but does not
+log the resulting payload after the Transform Message component because the Logger component consumes the stream.
+When the stream gets to the Transform Message component, the stream content is available in memory, so it is possible for Transform Message component to consume it. After the Transform Message component consumes the content, the second Logger receives an empty stream.
++
+.Mule 3 Streaming: Logging a Payload
+image::mruntime-streaming-about-2.png[]
++
```

Comment:
I think this example behaves differnetly. The first logger consumes the stream, so there is nothing left in the stream for the transformer to transform. Don't quite get what is the purpose of the second logger in this example.

Reply:
Plus, there is still mentions of the stream content being available in memory which is contradictory with the intent of this doc.

Reply:
Hi @elrodro83, I took this existing example from the Mule 4.3 Streaming doc, since it compares Mule 3 vs Mule 4 https://docs.mulesoft.com/mule-runtime/4.3/streaming-about
If this example is entirely wrong, we need to update it there as well... not sure who created it.

Since this is for the section _data stream can not be consumed more than once_ and the example mentions: _you try to log the payload before and
after a DataWeave transformation. The flow logs the payload before the Transform Message component, but does not
log the resulting payload after the Transform Message component because the Logger component consumes the stream. <--here indicates that the first Logger consumed the stream.

But then mentions the following sentence which is kind of contradictory because the stream which was already consumed by the first Logger can not get to the Transform component, is that right?

_When the stream gets to the Transform Message component, the stream content is available in memory, so it is possible for Transform Message to consume it. After Transform Message consumes the content, the second Logger receives an empty stream._  Should we remove these sentences and leave only the first part?
Or do we require a new brand example?

Thanks for your help.

Reply:
ok, i think the example is incomplete/not clear. The statements would be true only if the first logger doesn't consume the stream, but there is no mention to what that logger is logging. I assumed is was the same as the other logger in the flow, logging the payload. So, either remove the first logger from the example, which only adds confusion.

Reply:
@elrodro83 I removed the first Logger. Would the next text for the example be valid?
_In the following example, something similar happens when you try to log the payload after a DataWeave transformation. The HTTP Listener operation receives the payload stream, and then when the stream gets to the Transform Message component, it is available in memory, so the component consumes the stream. After the Transform Message component consumes the content, the second Logger receives an empty stream._

Reply:
that looks better